test/regression/sql/domain.sql

```diff
@@ -0,0 +1,53 @@
+create domain domainvarchar varchar(5);
+create domain domainnumeric numeric(8,2);
+create domain domainint4 int4;
+create domain domaintext text;
```

Comment:
Domains are often use to encode check constraints:

```
> CREATE DOMAIN positiveint int CHECK(VALUE > 0);
CREATE DOMAIN
> select 5::positiveint;
 positiveint
─────────────
           5
> select (-5)::positiveint;
ERROR:  23514: value for domain positiveint violates check constraint "positiveint_check"
SCHEMA NAME:  public
DATATYPE NAME:  positiveint
CONSTRAINT NAME:  positiveint_check
```

In the tests for this PR you don't use that feature. I'm also not sure if/how we can implement those checks. How about for this first version we error of the domain has a constraint?

Reply:
Okay, sorry about this. Looking at this again, I now think that it's fine to support domains with CHECK constraints too. Afaict any operation on a domain will have as the underlying type, not the domain. So DuckDB does not need to know about the check constraints. It would be good to document that in the limitations section of `docs/types.md` that DuckDB execution strips a DOMAIN its domain type and will convert it to its underlying type.

As an example of how Postgres operations strip the domain:
```
> create domain myint int check (value < 5);
> select 3::myint + 4::myint;
 ?column?
──────────
        7
(1 row)
> select pg_typeof(3::myint + 4::myint);
 pg_typeof
───────────
 integer
(1 row)
> select pg_typeof(sum(3::myint));
 pg_typeof
───────────
 bigint
(1 row)
```